File: frontend/src/types/budget.ts
export interface Budget {
  id: number
  accountId: number
  category: string
  amount: number
  startDate: string // "YYYY-MM-DD"
  endDate: string   // "YYYY-MM-DD"
  spent: number
}

export interface BudgetForm {
  category: string
  amount: number | null
  accountId: number | null
  startDate: string // "YYYY-MM-DD"
  endDate: string   // "YYYY-MM-DD"
}

export interface CreateBudgetResponse {
  message: string
  budget: Budget
}

export interface GetBudgetsResponse {
  budgets: Budget[]
}

export interface DeleteBudgetResponse {
  message: string
}

File: backend/internal/dto/budget.go
package dto

import (
	"backend/internal/models"
	
	"time"
)

type BudgetInput struct {
	Amount    float64
	Category  string
	AccountID uint
	StartDate time.Time
	EndDate   time.Time
}

type BudgetFilter struct {
	Status    string
	Year      *int
	Month     *int
	AccountID *uint
}

type BudgetResponse struct {
	ID        uint      `json:"id"`
	AccountID uint      `json:"accountId"`
	Category  string    `json:"category"`
	Amount    float64   `json:"amount"`
	StartDate time.Time `json:"startDate"`
	EndDate   time.Time `json:"endDate"`

	Spent     float64 `json:"spent"`
	Remaining float64 `json:"remaining"`
	IsActive  bool    `json:"isActive"`

	Account models.Account `json:"account"`
}

func ToBudgetResponse(
	budget models.Budget,
	spent float64,
	isActive bool,
) BudgetResponse {
	return BudgetResponse{
		ID:        budget.ID,
		AccountID: budget.AccountID,
		Category:  budget.Category,
		Amount:    budget.Amount,
		StartDate: budget.StartDate,
		EndDate:   budget.EndDate,

		Spent:     spent,
		Remaining: budget.Amount - spent,
		IsActive:  isActive,

		Account: budget.Account,
	}
}